test/cri-containerd/stopcontainer_test.go

```diff
@@ -148,3 +148,52 @@ func Test_StopContainer_WithExec_LCOW(t *testing.T) {
 		Stdout: true,
 	})
 }
+
+func Test_StopContainer_ReusePod_LCOW(t *testing.T) {
+	pullRequiredLcowImages(t, []string{alpineAspNet, alpineAspnetUpgrade})
+
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sandboxRequest := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name() + "-Sandbox",
+				Namespace: testNamespace,
+			},
+		},
+		RuntimeHandler: lcowRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, ctx, sandboxRequest)
+	defer removePodSandbox(t, client, ctx, podID)
+	defer stopPodSandbox(t, client, ctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		PodSandboxId: podID,
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: alpineAspNet,
+			},
+			Command: []string{
+				"top",
+			},
+		},
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	containerID := createContainer(t, client, ctx, request)
+	startContainer(t, client, ctx, containerID)
+	stopContainer(t, client, ctx, containerID)
+	removeContainer(t, client, ctx, containerID)
```

Comment:
Do we need any kind of checks here to see if any of these functions failed?

Reply:
Yes, I think you'll want to change this to make a call to a function like [here](https://github.com/microsoft/hcsshim/blob/432e537acbeb74fed644b5b34a781b2675cae9ab/test/cri-containerd/container_test.go#L59). This ensures we still clean up the container and any state in the event of a panic.

Reply:
good call